packages/vm/src/runBlock.ts

```diff
@@ -160,10 +160,10 @@ export async function runBlock(vm: VM, opts: RunBlockOpts): Promise<RunBlockResu
     // Populate the execution witness
     stateManager.initVerkleExecutionWitness!(block.header.number, block.executionWitness)
 
-    if (stateManager instanceof StatelessVerkleStateManager) {
+    if ('verifyPostState' in stateManager) {
```

Comment:
I think it would be nice to have a code comment directly on the original `verifyPostState` method how the method is (mis-)used and that it should therefore not be renamed.